src/cdk/a11y/id-generator.ts

```diff
@@ -35,6 +36,6 @@ export class _IdGenerator {
       counters[prefix] = 0;
     }
 
-    return `${prefix}${counters[prefix]++}`;
+    return `${prefix}${randomInfix ? this._infix + '-' : ''}${counters[prefix]++}`;
```

Comment:
super nit: I'm wondering if we should add the random number as a suffix. I remember that back in the day IDs starting with a number were problematic and in theory the `prefix` here can be an empty string.